### PR TITLE
Fixed minimal and maximal value of grouped product

### LINKED ISSUE #4336: 'Minimal value is 1!' and 'Maximum value is 5!' shown under the counter in grouped product
**Description**:
'Minimal value is 1!' shown under the counter in grouped product if user select 0 product of this type
'Maximum value is 5!' shown under the counter in grouped product if user select 6+ product of this type

**Expected behavior**:

- Messages: 'Minimal value is 1!' and 'Maximum value is 5!' - absent
- User can select 0 items of option with maximum Qty (Config Sunglasses-Large size) in grouped product 
- if qty increased to 5 then button + blocked same as for product 'Honor Smart Watch Min / Max Qty'

**Screenshots, Video, Logs**
image

**To Reproduce**:
1. Open https://master.d25kd40decjpue.amplifyapp.com/default/grouped-sunglasses.html
2. Click on - in front of 'Config Sunglasses-Large size'
3. Click on + in front of 'Config Sunglasses-Large size'

**Actual result**:
'Minimal value is 1!' shown under the counter in grouped product if user select 0 product of this type
'Maximum value is 5!' shown under the counter in grouped product if user select 6+ product of this type

### PR DESCRIPTION
**Pull request by SalimMersally**
**Related issue(s):**
* Fixes #4336 

**Problem:**
* 'Minimal value is 1!' shown under the counter in grouped product if user select 0 product of this type
* 'Maximum value is 5!' shown under the counter in grouped product if user select 6+ product of this type

**In this PR:**
* The user is now able to select 0 till max for any product, and then the buttons are disabled, also Messages: 'Minimal value is 1!' and 'Maximum value is 5!' are removed

### CODE REVIEW

**Comment by carinadues** [comment]
Hey!

There's one thing that requires fixing:

Some products have min qty, while 0 indicates that they do not want to add that item to cart, so it is not added when set to 0. The allowed values should be 0 and min qty to max qty.

For example in this grouped product: "Config Sunglasses-Large size" has min qty = 2 and "Config Sunglasses-Medium size" has min qty = 5

If the product has a min qty set, let's say, 2: 
The person can choose 0, 2 or more.
The person can't choose 1.
If the person chooses 1, the message 'Minimal value is 2!' should show up and be well positioned in mobile.

**Review by riha112 — CHANGES_REQUESTED** [review]
1. If you add those rules directly into `Field` component/container it will affect all form fields, thus ignoring ''0''' everywhere.

2. There is much simpler solution for this. Instead of adding custom rules in `Field.component` and `Field.container` you can specify custom validation rule directly in `GroupedProductsItem.component`.
To do that you use `match` inside `validationRule` attribute.

So in this case you would:
**1st** remove min range check, as it will be done now from `match`:
from:
```js
...
range: {
   min,
   max
}
```
to:
```js
...
range: {
   max
}
```

2nd extend `getError` function that is binded to `match` attribute:
```js
    getError(value) {
        const {
            product = {}
        } = this.props;

        const valueNum = +value;

        if (!!valueNum && !getProductInStock(product)) {
            return __('Product is out of stock');
        }

        const min = getMinQuantity(product);

        if (valueNum !== 0 && valueNum < min) {
            return __('Minimal value is %s!', min);
        }

        return true;
    }
```

More about field/form validation can be found here: https://www.notion.so/scandiweb/5-1-0-Config-Bundle-Form-Validation-update-v1-0-5366e403c9644a09ad634c5ffaada7cf#1844715db672410ea2a054889eb264b2

**Comment by Elena-Evsa** [comment]
1. If user opened PDP of grouped product with child product (max Qty = 5, default Qty = 4) then message 'Minimal value is 5!' should be shown immediately https://vimeo.com/686641829/3d98c94949. Now message shown after click on 'Add to cart' or after changing of Qty

File: packages/scandipwa/src/component/Field/Field.container.js [hunk 1]
```diff
@@ -79,6 +79,10 @@ export class FieldContainer extends PureComponent {
 
     fieldRef = createRef();
 
+    componentDidMount() {
+        this.validate();
+    }
```
Comment (riha112) [inline packages/scandipwa/src/component/Field/Field.container.js:84]: We have some form fields, that shouldn't be validated on load, so this solution won't work.

Instead change in **GroupedProductsItem.component.js**:
from: `validateOn={ ['onChange', 'onload'] }`
to: `validateOn={ ['onChange', 'onLoad'] }`
_(there was typo mistake for "onLoad" (capital L) event, thus it wasn't executing on load)_

**Review by riha112 — CHANGES_REQUESTED** [review]

**Review by riha112 — APPROVED** [review]

**Pull request merged**